lua/luasnip/init.lua

```diff
@@ -225,6 +244,17 @@ local function _jump_into_default(snippet)
 	return util.no_region_check_wrap(snippet.jump_into, snippet, 1)
 end
 
+---@class Snippet
```

Comment:
Oh, this gave me a few thoughts:
* I think the annotations should only communicate official API, which for a snippet means relatively few fields, I think `trigger`, `name`, `description`, `captures` and `env` would be official, or are so commonly-used that they won't be changed, while stuff like `id` and `wordTrig` should not (yet) be relied upon.
Also, there are a few interesting functions for snippet, see [here](https://github.com/L3MON4D3/LuaSnip/blob/master/DOC.md#api-1)
* This is kind of a weird place to define `Snippet`, we should probably use a separate file for basic classes like that (see `lua/luasnip/_types.lua` or `lua/luasnip/extras/_extra_types.lua` for examples of this) (also, looking at them, it may make sense to namespace our classes :sweat_smile:)
* Not sure if you recall my little digression on snippet vs expandable vs addable in the discussion you opened, but `snip_expand` would take an `expandable`. But since this distinction exists in very few places, and is yet another thing newcomers would have to deal with, it's probably fine, especially for now, to just make most things `Snippet`

Reply:
(correction: `id` can also be official :sweat_smile: Just saw that we have `get_id_snippet`, and that'd be pretty stupid without some way to get the `id`)